### DISCUSSION TITLE
Topmost In Linux

### DISCUSSION THREAD

**Discussion by sadlikkj** [AvaloniaUI/Avalonia#15325] [Q&A]
Has anyone run into the situation where when Topmost is set to true, the application doesn't obey the setting? The application does not stay Topmost if navigating to a different window. The error occurs using a Debian 12 Linux distribution and works appropriately on Windows 10. I'm tempted to utilize the Deactivated event to hide and show the window to force a Topmost but with varying success.

**Comment by kekekeks** [comment]
Note that modern window managers are actively trying to prevent your window from staying on top if they think that the user doesn't want that. Also note, that "topmost", is just a hint for a window manager.

**Reply by kekekeks** [reply]
Which is also why it's so important to NOT have a custom window frame: default one allows the user to configure "topmost" behavior on e. g. KDE.

**Reply by sadlikkj** [reply]
Thank you for such a quick reply!

My situation requires topmost as it's an application that I provide to a 3rd party, which communicates via gRPC and is intended to be the sole application for user interaction. Do you happen to have a suggestion of how to achieve this in Linux or am I SOL and need to think of another solution?

**Reply by kekekeks** [reply]
If it's a kiosk-like env, I'd suggest using a kiosk-specialized wayland compositor like [cage](https://github.com/cage-kiosk/cage). GNOME desktop environment isn't really friendly to apps attempting to take more control over the desktop.

**Reply by thevortexcloud** [reply]
Similarly, if you don't need a desktop you can also achieve this by using DRM with Avalonia.
